Readline-Release: 7.0
Patch-ID: readline70-004

Bug-Reported-by:        Kieran Grant <kieran.thehacker.grant@gmail.com>
Bug-Reference-ID:       <ec9071ae-efb1-9e09-5d03-e905daf2835c@gmail.com>
Bug-Reference-URL:      http://lists.gnu.org/archive/html/bug-bash/2018-02/msg00002.html

Bug-Description:

With certain values for PS1, especially those that wrap onto three or more
lines, readline will miscalculate the number of invisible characters,
leading to crashes and core dumps.
Index: b/display.c
===================================================================
--- a/display.c
+++ b/display.c
@@ -771,7 +771,9 @@ rl_redisplay ()
 	 appear in the first and last lines of the prompt */
       wadjust = (newlines == 0)
 		  ? prompt_invis_chars_first_line
-		  : ((newlines == prompt_lines_estimate) ? wrap_offset : prompt_invis_chars_first_line);
+		  : ((newlines == prompt_lines_estimate)
+		  	? (wrap_offset - prompt_invis_chars_first_line)
+		  	: 0);
 
       /* fix from Darin Johnson <darin@acuson.com> for prompt string with
          invisible characters that is longer than the screen width.  The
Index: b/patchlevel
===================================================================
--- a/patchlevel
+++ b/patchlevel
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-3
+4
